web/templates/recipient_row.hbs

```diff
@@ -35,7 +35,7 @@
                 {{/if}}
             </a>
         </span>
-        <span class="recipient_bar_controls no-select">
+        <span class="recipient_bar_controls no-select" data-stream-id="{{stream_id}}" data-topic-name="{{topic}}" data-topic-url="{{topic_url}}">
```

Comment:
Let's not duplicate these attributes here -- it will slow down interleaved views to have a bunch of duplicate data in the HTML. Any code that needs them can get them from the top-level `recipent_row`. 

The URL you can always compute from the stream/topic use `url_encoding.ts` functions, so there's no need to have a `data-topic-url` here either. (I think the cases where we do a `data-topic-url` are not in loops like this code path is).

Reply:
I haved the attributes to the ".message_header" element, which is one element deeper than the `.recipient_row` element as we are confirmed that the message group belongs to a stream and not DMS at that level:
```handlebars
{{#if is_stream}}
<div class="message_header message_header_stream right_part" data-stream-id="{{stream_id}}" {{#if topic}}data-topic-name="{{topic}}"{{/if}}>
```